### ISSUE TITLE
Allow to check if a document was updated again

### ISSUE THREAD

**Issue by Sorokin-Anton** [c-cube/linol#26]
Imagine it takes 1 second to process changes in a document (e.g. create diagnostics) and a user edits file 50 times in 5 seconds (i.e. he did 50 keystrokes). Then the thing hooked to `TextDocumentDidUpdate` will be sequentially called 50 times, and user will see the actual diagnostics only over 50 seconds.

I'd like to have a way to "look ahead" if there were a new document-update requests for my doc before calculating the diagnostics (If yes  -I'll just skip checfking current code since its outdated)

**Comment by Sorokin-Anton** [comment]
@c-cube I don't have a clear idea how to solve this problem on linol level (AFAIU it uses lwt channels for data, and there is no lookahead for channels), and I was unable to create a working concurrent solution (probably it's related to the fact that I have to use ocaml 4.14 not the ocaml 5), so any suggestions are welcome

**Comment by Sorokin-Anton** [comment]
cc @heitor-lassarote @XJIE6

**Comment by c-cube** [comment]
yes, I use a "switch" like construct to cancel previous computations when a document is updated. I think `Lwt_switch.t` can do it.

**Comment by Gbury** [comment]
> yes, I use a "switch" like construct to cancel previous computations when a document is updated. I think `Lwt_switch.t` can do it.

Sure, but that only works if your computation already reasonably and regularly encounters lwt binds, no ? Since Lwt will only be able to cancel a computation at these bind points, you cannot interrupt a pure ocaml non-lwt computation.

**Comment by c-cube** [comment]
Sure, there's no surefire way. You can check on the switch regularly from inside your computation though.

**Comment by Sorokin-Anton** [comment]
Well, I want to launch heavy external function, so it's not that easy. I still think there should be a way to create a buffer with next received request before processing the first request in current queue

**Comment by Sorokin-Anton** [comment]
@Gbury about delay: it's ok to have delay, but I still need to know somehow if new requests have arrived during this delay

**Comment by Gbury** [comment]
Indeed, and it might be reasonable, at least from inside `linol`.
